modules/payment/src/Plugin/PaymentMethod/PaymentMethodBase.php

```diff
@@ -139,7 +139,7 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
    */
   public function buildPaymentForm(array $pane_form, FormStateInterface $form_state, array &$complete_form) {
     $pane_form['cc_number'] = [
-      '#type' => 'textfield',
+      '#type' => 'number',
```

Comment:
@sdelbosc bit concerned about this, it will start showing steps (up/down arrows and up down keys will start changing value)

Reply:
check with Mitesh how this was done on Alshaya, maybe some JS should come along the way

Reply:
@sdelbosc Thanks. @nikunjkotecha is right. I missed that part, I thought css would resolve that. But, I didn't take keyboard behaviour in account. so, either we need to change the type to `tel` OR create a custom element type for `CVV`.

Reply:
Go on the discussion on our project PR first we will update here once done

Reply:
@miteshmap Do you have an update on this discussion?

Reply:
@malachy-mcconnell yes, we used type "tel" and worked as expected for mobile.